src/borg/_hashindex.c

```diff
@@ -376,31 +377,69 @@ hashindex_get(HashIndex *index, const void *key)
     return BUCKET_ADDR(index, idx) + index->key_size;
 }
 
+inline
+int
+distance(HashIndex  *index, int current_idx, int ideal_idx)
+{
+    /* If the current index is smaller than the ideal index we've wrapped
+       around the end of the bucket array and need to compensate for that. */
+    return current_idx - ideal_idx + ( (current_idx < ideal_idx) ? index->num_buckets : 0 );
+}
+
 static int
 hashindex_set(HashIndex *index, const void *key, const void *value)
 {
     int idx = hashindex_lookup(index, key);
-    uint8_t *ptr;
+    uint8_t *bucket_ptr;
+    int offset = 0;
+    int other_offset;
+    void *bucket = malloc(index->key_size + index->value_size);
+    void *buffer = malloc(index->key_size + index->value_size);
     if(idx < 0)
     {
+        /* we don't have the key in the index
+           we need to find an appropriate address */
```

Comment:
I am not sure whether the C code follows some specific coding style, but the max line length for Python code in borg is 120, thus such a comment would be just a single line.